substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/monitor/MultiThreadedMonitorSupport.java

```diff
@@ -255,8 +257,27 @@ private static void slowPathMonitorEnter(Object obj) {
     @RestrictHeapAccess(reason = NO_LONGER_UNINTERRUPTIBLE, access = Access.UNRESTRICTED)
     @Override
     public void monitorEnter(Object obj) {
+        long startTicks = com.oracle.svm.core.jfr.JfrTicks.elapsedTicks();
```

Comment:
HotSpot only emits the event if there is a contention, i.e., no JFR event is emitted for the fast path. We need something similar.

Reply:
I've now implemented this behavior by only emitting events if tryLock() fails to get the lock

Reply:
This new approach doesn't work:
1. You only fill `monitorOwners` when a lock contention is happening. So, you will usually keep track of an outdated value and not of the latest previous owner. One possible solution would be to introduce a custom `ReentrantLock` subclass that keeps track of the previous owner in a field. Writing the JFR thread id to that field should be cheap enough so that you can do it in the fast path.
2. There is no guarantee that your call to `lockObject.lock();` really runs into a contention. So, you will emit unnecessary JFR events from time to time. To solve that, you will probably need to add a method that is similar to `lock()` that returns if it was necessary to park the thread because of a contention.

Reply:
yes I see! Thank you Christian. I have used the solution you suggested for **1**. 

For **2**, wouldn't `tryLock()` returning false indicate that there is contention? Is the problem that the lock may be released in between the call to `tryLock()` and `lock()`? I think the Hotspot code makes one attempt to acquire the lock (+1 optional spin) and upon failure treats it as contention.